lib/themes/dosomething/paraneue_dosomething/js/campaign/Controller.js

```diff
@@ -0,0 +1,75 @@
+import setting from '../utilities/Setting';
+import ApiClient from '../utilities/ApiClient';
+import { sortCampaignsByRelevance } from '../campaign/helpers';
+
+class CampaignController {
+  constructor() {
+    this.campaignsToExclude = setting('dsUser.activity.signups', []);
+    this.campaignsApiPage = 1;
+    this.apiClient = new ApiClient('v1');
+    this.retrievedCampaigns = [];
+  }
+
+  filter(campaigns, comparison) {
+    return campaigns.filter(campaign => comparison.indexOf(parseInt(campaign.id)) == -1);
+  }
+
+  /**
+   * Grab campaigns & alter based on given args.
+   * @param {Object} args containing the following optional parameters
+   *   - {boolean} isStaffPick - Defaults to false.
+   *   - {int} page - Defaults to 1.
+   *   - {boolean} removeDuplicateResults
+   *   - {boolean} sortByRelevance
+   *   - {boolean} removeSignups
+   * @return {Promise}
+   */
+  getCampaigns(args) {
+    const query = {
+      staff_pick: args['isStaffPick'] ? +args['isStaffPick'] : 0, // API requires 1/0 instead of true/false
+      page: args['page'] || 1,
+    };
+
+    return this.apiClient.get('campaigns', query).then(response => {
```

Comment:
Yeah, I forgot we're in the `getCampaigns()` method which should return the data. Alternatively you could put all the logic inside that anonymous function inside a separate method like `processCampaigns()` or something, and then just call that from within the anonymous function? Not a huge deal though.